### ISSUE TITLE
macOS: error: while setting up the build environment: cannot unlink `/nix/store/...`: Permission denied

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nix#1504]
Using the new multi-user setup:

```
building path(s) ‘/nix/store/nwsjbvsw24q60lcpwrzpj6gj1l1ash28-SHA-1.6.4.2’
error: while setting up the build environment: cannot unlink ‘/nix/store/506xj88sv7m0i5fxrp789zfyl7b2iylx-SHA-1.6.4.2.drv.sb’: Permission denied
```

```
$ nix-env --version
nix-env (Nix) 1.11.12
```

cc @NixOS/darwin-maintainers

**Comment by domenkozar** [comment]
@Ericson2314 is this an instance of https://github.com/NixOS/nix/pull/1482 ?

**Comment by copumpkin** [comment]
I've seen that "cannot unlink *.sb" error intermittently but assumed it was something on my machine. No idea what's going on with it, unfortunately. The .sb files are the "seatbelt" specs for Darwin sandboxes, if that helps.

**Comment by Ericson2314** [comment]
https://github.com/NixOS/nix/commit/c740c3ce500af2b7eb34651b5eeec01288d79dca Nothing to do with me, thankfully.

**Comment by davidak** [comment]
I got this also multiple times today.

```
imac:nixpkgs davidak$ nix-env -f ~/code/nixpkgs -i iperf
installing ‘iperf-3.2’
[...]
these paths will be fetched (0.56 MiB download, 0.56 MiB unpacked):
  /nix/store/5a6ah56xs24015rfan5cpy5jvnwb15p7-iperf-3.2.tar.gz
fetching path ‘/nix/store/5a6ah56xs24015rfan5cpy5jvnwb15p7-iperf-3.2.tar.gz’...
building path(s) ‘/nix/store/b2h5jvfnbjr6s24mq8vyrc9cf1x122hb-llvm-4.0.1’, ‘/nix/store/r3il51p7zl4ydg5wyvkkii5s857ba9m4-llvm-4.0.1-lib’
killing process 15064
error: while setting up the build environment: cannot unlink ‘/nix/store/ck77ymaf94wihs55f2sfzgjrnmxj7fcv-llvm-4.0.1.drv.sb’: Permission denied
```

```
make[2]: Entering directory '/private/tmp/nix-build-diffutils-3.6.drv-0/diffutils-3.6/lib'
  CC       allocator.o
yes
checking whether fseeko is declared without a macro...   CC       areadlink.o
building path(s) ‘/nix/store/5bkanas7adwyx74s4zzrh9vpgm0lxf0a-expand-response-params’
error: while setting up the build environment: cannot unlink ‘/nix/store/zdlf3bc29xdl1gz9driwhvzc5shp9m8x-expand-response-params.drv.sb’: Permission denied
imac:nixpkgs davidak$ ll /nix/store/zdlf3bc29xdl1gz9driwhvzc5shp9m8x-expand-response-params.drv.sb
-rw-r--r--  1 nixbld5  nixbld  54  7 Aug 21:18 /nix/store/zdlf3bc29xdl1gz9driwhvzc5shp9m8x-expand-response-params.drv.sb
```

---
OS X 10.10.5
nix-env (Nix) 1.11.13

**Comment by Ericson2314** [comment]
This happens when a build fails, and then the same build is attempted again.

**Comment by domenkozar** [comment]
Yes :) The problem is, if you build with `-j` then you are in constant loop, as you `nix-store --delete` one broken packages but when paralell build breaks again, there are multiple broken folders again.

**Comment by domenkozar** [comment]
Partialy leads to @shlevy in 439014231

**Comment by domenkozar** [comment]
Aha already fixed in master: c740c3ce500af2b7eb34651b5eeec01288d79dca

**Comment by domenkozar** [comment]
Also fixed in 1.11.15
